base_station/gui/src/components/ToggleButton.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+    <label :for="id + '_button'" :class="{'active': isActive}" class="toggle__button">
+        <span v-if="isActive" class="toggle__label">{{ enableText }}</span>
+        <span v-if="! isActive" class="toggle__label">{{ disabledText }}</span>
+
+        <input type="checkbox" :disabled="disabled" :id="id + '_button'" v-model="checkedValue">
+        <span class="toggle__switch"></span>
+    </label>
+</template>
+
+<script>
+    export default {
+    props: {
+        disabled: {
+            type: Boolean,
+            default: false
+        },
+        labelEnableText: {
+            type: String,
+            default: 'On'
+        },
+        
+        labelDisableText: {
+            type: String,
+            default: 'Off'
+        },
+        id: {
+            type: String,
+            default: 'primary'
+        }, 
+        defaultState: {
+            type: Boolean,
+            default: false
+        }
+    },
+    data() {
+        return {
+            currentState: this.defaultState
+        }
+    },
+    watch: {
+        defaultState: function defaultState() {
```

Comment:
I don't think it makes sense to have defaultState be watched. Theoretically this value should never change. However, I think you could currentState a prop passed in from the parent component instead of a local data member. In that case, you could just get rid of defaultState all together. The parent component would set currentState to a variable (like scoopUVActive) which has its own default value and can be changed by the parent